### ISSUE TITLE
if PYRIGHT: conditional statement

### ISSUE THREAD

**Issue by abelcheung** [microsoft/pyright#4734]
**Is your feature request related to a problem? Please describe.**
When a project tries to support multiple type checkers, it is unavoidable that different checkers don't always support identical features. In many cases, people opt for using the lowest common denominator, but this approach prevents people from trying new features (such as draft PEPs).

**Describe the solution you'd like**
Mypy has a similar construct:

```python
if MYPY:
    ...
else:
    ...
```

I think mypy just wants to make it a synonym of `if TYPE_CHECKING`, but it can actually mean something different, if multiple type checkers are involved. This can be especially useful inside external stub files. Below snippet uses PEP ~698~ 696 as example:

```python
if MYPY:  # pyright: ignore[reportUndefinedVariable]
    _T = TypeVar('_T')
else:
    _T = TypeVar('_T', default=int) 
```

However the snippet only distinguishes between mypy and every other type checker. If there is certain feature only implemented by pyright, `if PYRIGHT` would be needed.

**Comment by erictraut** [comment]
Pyright already has an option for defining your own static constants. 

Simply add this to your pyrightconfig.json file:
```
"defineConstant": {
    "PYRIGHT": true
  },
```

Or if you're using a pyproject.toml:
```
defineConstant = { PYRIGHT = true }
```

**Comment by abelcheung** [comment]
@erictraut Ouch, apparently I haven't been reading doc enough. Thanks for the pointer!
